File: validator-app/src/main/java/org/cna/keyple/famoco/validator/Application.java
package org.cna.keyple.famoco.validator;

import org.cna.keyple.famoco.validator.di.DaggerAppComponent;

import dagger.android.AndroidInjector;
import dagger.android.DaggerApplication;

public class Application extends DaggerApplication {

    @Override
    protected AndroidInjector<? extends DaggerApplication> applicationInjector() {
        return DaggerAppComponent.builder().application(this).build();
    }
}

File: validator-app/src/main/kotlin/org/cna/keyple/famoco/validator/Application.kt
package org.cna.keyple.famoco.validator

import dagger.android.AndroidInjector
import dagger.android.DaggerApplication
import org.cna.keyple.famoco.validator.di.AppComponent
import org.cna.keyple.famoco.validator.di.DaggerAppComponent
import timber.log.Timber
import timber.log.Timber.DebugTree

class Application : DaggerApplication() {
    override fun onCreate() {
        super.onCreate()
        Timber.plant(DebugTree())
    }

    override fun applicationInjector(): AppComponent? {
        return DaggerAppComponent.builder().application(this).build()
    }
}